File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    
    # 약수 개수 구하는 함수
    def prime(n):
        cnt = 0                 # 약수의 개수
        if n == 1:              # 숫자가 1이면 1리턴
            return 1
        if n == 2 or n == 3:    # 숫자가 2나 3이면 2 리턴
            return 2
        
        for i in range(1, int(n**(1/2)) + 1):   # 숫자 3부터 약수의 개수 구하기
            if n % i == 0:      # 약수 판별
                if i * i == n:  # 약수가 하나일 때, ex) 25 = 5 * 5
                    cnt += 1
                else:           # 약수가 짝지어져있을때, ex) 30 = 5 * 6
                    cnt += 2
            
            if cnt > limit:     # 약수의 개수가 제한수치보다 크면 공격력(power) 리턴
                return power
        
        return cnt              # 아니면 약수의 개수 리턴
            
    for n in range(1, number + 1):
        answer += prime(n)
    
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    
    function buying_weapon(num) {
        if (num === 1) {
            answer++;
            return
        }
        
        let count = 0;
        for (let i = 1; i <= num**(1/2); i++) {
            if (num % i === 0) {
                if (i * i === num) count++;
                else count += 2;
            }
        }
        
        answer += count <= limit ? count : power;
        return
    }
    
    for (let num = 1; num <= number; num++) buying_weapon(num)
    
    return answer;
}